x/auth/tx/config.go

```diff
@@ -27,25 +32,68 @@ type config struct {
 // NOTE: Use NewTxConfigWithHandler to provide a custom signing handler in case the sign mode
 // is not supported by default (eg: SignMode_SIGN_MODE_EIP_191). Use NewTxConfigWithTextual
 // to enable SIGN_MODE_TEXTUAL (for testing purposes for now).
-func NewTxConfig(protoCodec codec.ProtoCodecMarshaler, enabledSignModes []signingtypes.SignMode, customSignModes ...signing.SignModeHandler) client.TxConfig {
+//
+// We prefer to use depinject to provide client.TxConfig, but we permit this constructor usage.  Within the SDK,
+// this constructor is primarily used in tests, but also sees usage in app chains like:
+// https://github.com/evmos/evmos/blob/719363fbb92ff3ea9649694bd088e4c6fe9c195f/encoding/config.go#L37
+// TODO: collapse enabledSignModes and customSignModes
+func NewTxConfig(protoCodec codec.ProtoCodecMarshaler, enabledSignModes []signingtypes.SignMode,
+	customSignModes ...txsigning.SignModeHandler,
+) client.TxConfig {
+	// protoFiles should perhaps be a parameter to this function, but the choice was made here to not break the
+	// NewTxConfig API.
+	protoFiles := registry.MergedProtoRegistry()
+	typeResolver := protoregistry.GlobalTypes
+	signersContext, err := txsigning.NewGetSignersContext(txsigning.GetSignersOptions{ProtoFiles: protoFiles})
```

Comment:
This is unnecessary because of #15873. Just use `codec.InterfaceRegistry().SigningContext()`

Reply:
This is a compile error right now.  I think it is OK to land this in main first then work on #15873 (which is still failing tests) then refactor?